### PR TITLE
Use NIO when NIOTS isn't available

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

It's useful to have an example of how people can fall back to NIO when
NIOTS isn't available.

Modifications:

Add support to fallback to using NIO when NIOTS isn't available.

Result:

NIOSMTP can now run on iOS 10 and iOS 11 where NIOTS isn't available.

### CODE REVIEW

File: NIOSMTP/NIOSMTP.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -340,8 +344,9 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = U9N8BQB923;
```
Comment (weissi) [inline NIOSMTP/NIOSMTP.xcodeproj/project.pbxproj]: what's this team?

File: NIOSMTP/NIOSMTP/ViewController.swift [hunk 2]
```diff
@@ -104,3 +108,112 @@ class ViewController: UIViewController {
     }
 }
 
+// MARK: - NIO/NIOTS handling
+
+/// This protocol is intended as a layer of abstraction over `ClientBootstrap` and
+/// `NIOTSConnectionBootstrap`. We only need a `connect` method since configuration
+/// is done on the concrete type.
+public protocol ClientBootstrapProtocol {
```
Comment (weissi) [inline NIOSMTP/NIOSMTP/ViewController.swift]: does this need to be `public`?

File: NIOSMTP/NIOSMTP/ViewController.swift [hunk 3]
```diff
@@ -104,3 +108,112 @@ class ViewController: UIViewController {
...
+            MessageToByteHandler(SMTPRequestEncoder()),
+            SendEmailHandler(configuration: Configuration.shared.serverConfig,
+                             email: email,
+                             allDonePromise: emailSentPromise)
+        ]
+
+        if Configuration.shared.useTLS {
+            do {
+                let sslContext = try NIOSSLContext(configuration: .forClient())
+                let sslHandler = try NIOSSLClientHandler(context: sslContext, serverHostname: nil)
```
Comment (weissi) [inline NIOSMTP/NIOSMTP/ViewController.swift]: don't you need to fill in the `serverHostname` here?

File: NIOSMTP/NIOSMTP/ViewController.swift [hunk 4]
```diff
@@ -104,3 +108,112 @@ class ViewController: UIViewController {
...
+
+/// Network implementation and by extension which version of NIO to use.
+enum NetworkImplementation {
+    /// POSIX sockets and NIO.
+    case posix
+
+    #if canImport(Network)
+    @available(OSX 10.14, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+    /// Network.framework and NIOTransportServices.
+    case networkFramework
```
Comment (weissi) [inline NIOSMTP/NIOSMTP/ViewController.swift]: if you don't mind using `transportServices` here. That's just the term we use and we see Network.framework as an implementation of 'transport services'

File: NIOSMTP/NIOSMTP/ViewController.swift [hunk 5]
```diff
@@ -104,3 +108,112 @@ class ViewController: UIViewController {
...
+    #endif
+
+    /// Return the best implementation available for this platform, that is Network.framework and
+    /// NIOTransportServices when it is available or POSIX and NIO otherwise.
+    static var best: NetworkImplementation {
```
Comment (weissi) [inline NIOSMTP/NIOSMTP/ViewController.swift]: CC @neonacho 😂

File: NIOSMTP/NIOSMTP/ViewController.swift (hunk 4 above)
Comment (glbrntt) [reply NIOSMTP/NIOSMTP/ViewController.swift]: Ah makes sense, are you still okay with `NetworkImplementation` as the name of the `enum` though?

File: NIOSMTP/NIOSMTP/ViewController.swift (hunk 3 above)
Comment (glbrntt) [reply NIOSMTP/NIOSMTP/ViewController.swift]: yup - 🤦‍♂️

File: NIOSMTP/NIOSMTP/ViewController.swift (hunk 4 above)
Comment (weissi) [reply NIOSMTP/NIOSMTP/ViewController.swift]: @glbrntt I think that's fine.

**Pull request merged**